numba/cuda/simulator/cudadrv/devicearray.py

```diff
@@ -30,23 +30,66 @@ class FakeShape(tuple):
     indexing, similar to the shape in CUDA Python. (Numpy shape arrays allow
     negative indexing)
     '''
+
     def __getitem__(self, k):
         if isinstance(k, int) and k < 0:
             raise IndexError('tuple index out of range')
         return super(FakeShape, self).__getitem__(k)
 
 
+class FakeWithinKernelCUDAArray(object):
+    '''
+    Created to emulate the behavior of arrays within kernels, where either
+    array.item or array['item'] is valid. This behaviour does not follow
```

Comment:
Can we add a 
```suggestion
    Created to emulate the behavior of arrays within kernels, where either
    array.item or array['item'] is valid (that is, give all structured
    arrays `numpy.recarray`-like semantics). This behaviour does not follow
```

Reply:
isn't that implicit? What's the other case?

Reply:
It's implicit, but it's perhaps not obvious if you're not familiar in detail with structured arrays and recarrays - I think it's helpful to make the distinction for clarity.

Reply:
works for me

Reply:
The point I'd like to see is that the semantics aren't absent from numpy, they're just usually opt-in.